doc/api/esm.md

```diff
@@ -50,65 +50,17 @@ or when referenced by `import` statements within ES module code:
 * Files ending in `.cjs`.
 
 * Files ending in `.js` when the nearest parent `package.json` file contains a
-  top-level field `"type"` with a value of `"commonjs"`.
+  top-level field [`"type"`][] with a value of `"commonjs"`.
 
 * Strings passed in as an argument to `--eval` or `--print`, or piped to
   `node` via `STDIN`, with the flag `--input-type=commonjs`.
 
-### `package.json` `"type"` field
-
-Files ending with `.js` will be loaded as ES modules when the nearest parent
-`package.json` file contains a top-level field `"type"` with a value of
-`"module"`.
-
-The nearest parent `package.json` is defined as the first `package.json` found
-when searching in the current folder, that folder’s parent, and so on up
-until the root of the volume is reached.
-
-<!-- eslint-skip -->
-```js
-// package.json
-{
-  "type": "module"
-}
-```
-
-```sh
-# In same folder as above package.json
-node my-app.js # Runs as ES module
-```
-
-If the nearest parent `package.json` lacks a `"type"` field, or contains
-`"type": "commonjs"`, `.js` files are treated as CommonJS. If the volume root is
-reached and no `package.json` is found, Node.js defers to the default, a
-`package.json` with no `"type"` field.
-
-`import` statements of `.js` files are treated as ES modules if the nearest
-parent `package.json` contains `"type": "module"`.
-
-```js
-// my-app.js, part of the same example as above
-import './startup.js'; // Loaded as ES module because of package.json
-```
-
-Package authors should include the `"type"` field, even in packages where all
-sources are CommonJS. Being explicit about the `type` of the package will
-future-proof the package in case the default type of Node.js ever changes, and
-it will also make things easier for build tools and loaders to determine how the
-files in the package should be interpreted.
-
-Regardless of the value of the `"type"` field, `.mjs` files are always treated
-as ES modules and `.cjs` files are always treated as CommonJS.
-
 ### Package Scope and File Extensions
 
-A folder containing a `package.json` file, and all subfolders below that folder
-down until the next folder containing another `package.json`, is considered a
-_package scope_. The `"type"` field defines how `.js` files should be treated
```

Comment:
We still need this first sentence, that defines what a package scope is. I would perhaps keep this paragraph unchanged.

Reply:
Does the whole section on package scope and file extension belong into the `package.json` section? It doesn't seem to be specific to ESM.

Reply:
I have added links to the definition of package scope in package.json.html,  do you think that is sufficient?

> Does the whole section on package scope and file extension belong into the package.json section? It doesn't seem to be specific to ESM.

It is actually a sub-section of [Enabling](https://nodejs.org/docs/latest/api/esm.html#esm_enabling), and because CJS is default in node, it makes sense to have it documented here IMO.